NebulaHost/PacketProcessors/Factory/Entity/CreatePrebuildsRequestProcessor.cs

```diff
@@ -110,31 +110,44 @@ public void ProcessPacket(CreatePrebuildsRequest packet, NebulaConnection conn)
             }
         }
 
-        public bool CheckBuildingConnections(List<BuildPreview> buildPreviews, PlanetFactory factory)
+        public bool CheckBuildingConnections(List<BuildPreview> buildPreviews, EntityData[] entityPool, PrebuildData[] prebuildPool)
         {
             //Check if some entity that is suppose to be connected to this building is missing
             for(int i = 0; i < buildPreviews.Count; i++)
-            {
-                bool isInputOk = true;
-                if (buildPreviews[i].inputObjId > 0)
-                {
-                    isInputOk = factory.entityPool.Length >= buildPreviews[i].inputObjId && factory.entityPool[buildPreviews[i].inputObjId].id != 0;
-                } else if (buildPreviews[i].inputObjId < 0)
-                {
-                    isInputOk = factory.prebuildPool.Length >= -buildPreviews[i].inputObjId && factory.prebuildPool[-buildPreviews[i].inputObjId].id != 0;
-                }
-                bool isOutputOk = true;
-                if (buildPreviews[i].outputObjId > 0)
-                {
-                    isInputOk = factory.entityPool.Length >= buildPreviews[i].outputObjId && factory.entityPool[buildPreviews[i].outputObjId].id != 0;
```

Comment:
See here, isOutputOk should've been set. I've refactored it a bit, makes it easier to follow.

Especially since `factory.entityPool.Length == 3` means `0,1,2` are valid, but this allows `entityPool[3]` to be considered valid.

Reply:
I agree, this looks much better.